### PR TITLE
Raise an error for issue #1339

### PR DESCRIPTION
**Pull request by EmilyBourne**
Raise an error indicating that C should be used to mix scalar and array return types (see #1339).

### CODE REVIEW

File: pyccel/codegen/printing/fcode.py [hunk 1]
```diff
@@ -1875,15 +1875,28 @@ def _print_LiteralString(self, expr):
...
-        if expr.functions[0].is_inline:
+        if example_func.is_inline:
             return ''
 
+        if len(example_func.results) == 1:
+            if len(set(f.results[0].var.rank == 0 for f in interface_funcs)) != 1:
+                message = ("Fortran cannot yet handle a templated function returning either a scalar or an array "
+                           "If you are using the terminal interface, please pass --language c, "
+                           "if you are using the interactive interfaces epyccel or lambdify, please pass language='c' . "
+                           "See https://github.com/pyccel/pyccel/issues/1339 to see the advancement of this issue.")
```
Comment (yguclu) [inline pyccel/codegen/printing/fcode.py:1891]: Repeated "see"

Comment (EmilyBourne) [reply pyccel/codegen/printing/fcode.py:1891]: What do you suggest instead?

Comment (EmilyBourne) [reply pyccel/codegen/printing/fcode.py:1891]: monitor?

Comment (yguclu) [reply pyccel/codegen/printing/fcode.py:1891]: Fancy

Comment (EmilyBourne) [reply pyccel/codegen/printing/fcode.py]: Done : 85f8d24c4

File: pyccel/codegen/printing/fcode.py [hunk 2]
```diff
@@ -1875,15 +1875,28 @@ def _print_LiteralString(self, expr):
...
-        if expr.functions[0].is_inline:
+        if example_func.is_inline:
             return ''
 
+        if len(example_func.results) == 1:
+            if len(set(f.results[0].var.rank == 0 for f in interface_funcs)) != 1:
+                message = ("Fortran cannot yet handle a templated function returning either a scalar or an array "
+                           "If you are using the terminal interface, please pass --language c, "
+                           "if you are using the interactive interfaces epyccel or lambdify, please pass language='c' . "
+                           "See https://github.com/pyccel/pyccel/issues/1339 to monitor the advancement of this issue.")
```
Comment (yguclu) [inline pyccel/codegen/printing/fcode.py:1891]: Punctuation:
```suggestion
                message = ("Fortran cannot yet handle a templated function returning either a scalar or an array. "
                           "If you are using the terminal interface, please pass --language c, "
                           "if you are using the interactive interfaces epyccel or lambdify, please pass language='c'. "
                           "See https://github.com/pyccel/pyccel/issues/1339 to monitor the advancement of this issue.")
```

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good to me!

**Review by yguclu — APPROVED** [review]

**Pull request merged**